packages/eslint-plugin/src/rules/no-redundant-type-constituents.ts

```diff
@@ -203,15 +417,18 @@ export default createRule({
     messages: {
       errorTypeOverrides: `'{{typeName}}' is an 'error' type that acts as 'any' and overrides all other types in this {{container}} type.`,
       literalOverridden: `{{literal}} is overridden by {{primitive}} in this union type.`,
+      objectOverridden: `{{subType}} is overridden by {{superType}} in this union type.`,
       overridden: `'{{typeName}}' is overridden by other types in this {{container}} type.`,
       overrides: `'{{typeName}}' overrides all other types in this {{container}} type.`,
       primitiveOverridden: `{{primitive}} is overridden by the {{literal}} in this intersection type.`,
```

Comment:
[Feature] Is there a need to keep differentiating between literal and primitive types? I would think the new logic should still apply to them, right?

Suggestion: have you considered unifying the existing & new logic into one area of checks? It'd have to be careful not to try to compare keys of literals or primitives, but otherwise I think it should all be roughly the same, right?

Thinking out loud: I don't think there's a strong need to keep differentiating between `intersection` and `union` in the message either...